services/sqs/src/test/java/software/amazon/awssdk/services/sqs/SqsBatchManagerTest.java

```diff
@@ -0,0 +1,389 @@
+            + "    <BatchResultErrorEntry>\n"
+            + "        <Id>%s</Id>\n"
+            + "        <Code>%s</Code>\n"
+            + "        <Message>%s</Message>\n"
+            + "    </BatchResultErrorEntry>\n"
+            + "</SendMessageBatchResult>\n"
+            + "</SendMessageBatchResponse>", id1, errorCode, errorMessage, id2, errorCode, errorMessage);
+
+        stubFor(any(anyUrl()).willReturn(aResponse().withStatus(200).withBody(responseBody)));
+        List<CompletableFuture<SendMessageResponse>> responses = createAndSendSendMessageRequests(id1, id2);
+
+        SendMessageResponse completedResponse1 = responses.get(0).join();
+        SendMessageResponse completedResponse2 = responses.get(1).join();
+        String expectedHash = getMd5Hash(String.format("%s: %s", errorCode, errorMessage));
+        assertThat(completedResponse1.md5OfMessageBody()).isEqualTo(expectedHash);
+        assertThat(completedResponse2.md5OfMessageBody()).isEqualTo(expectedHash);
+    }
+
+    @Test
+    public void sendMessageBatchFunctionReturnsWithError_completeMessagesExceptionally() {
+        String id1 = "0";
+        String id2 = "1";
+        String responseBody = "<Error>\n"
+                              + "<Code>CustomError</Code>\n"
+                              + "<Message>Foo bar</Message>\n"
+                              + "<RequestId>656c76696e6727732072657175657374</RequestId>\n"
+                              + "<HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>\n"
+                              + "</Error>";
+
+        stubFor(any(anyUrl()).willReturn(aResponse().withStatus(400).withBody(responseBody)));
+        List<CompletableFuture<SendMessageResponse>> responses = createAndSendSendMessageRequests(id1, id2);
+
+        CompletableFuture<SendMessageResponse> response1 = responses.get(0);
+        CompletableFuture<SendMessageResponse> response2 = responses.get(1);
+        assertThatThrownBy(response1::join).isInstanceOf(CompletionException.class).hasMessageContaining("400");
+        assertThatThrownBy(response2::join).isInstanceOf(CompletionException.class).hasMessageContaining("400");
+    }
+
+    @Test
+    public void sendMessageBatchNetworkError_causesConnectionResetException() {
+        String id1 = "0";
+        String id2 = "1";
+        stubFor(any(anyUrl()).willReturn(aResponse().withFault(Fault.CONNECTION_RESET_BY_PEER)));
+
+        List<CompletableFuture<SendMessageResponse>> responses = createAndSendSendMessageRequests(id1, id2);
+
+        CompletableFuture<SendMessageResponse> response1 = responses.get(0);
+        CompletableFuture<SendMessageResponse> response2 = responses.get(1);
+        assertThatThrownBy(response1::join).isInstanceOf(CompletionException.class).hasMessageContaining("Connection reset");
+        assertThatThrownBy(response2::join).isInstanceOf(CompletionException.class).hasMessageContaining("Connection reset");
```

Comment:
It seems this test failed on JDK11.
```
Expecting
<"software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: Read timed out">
to contain:
"Connection reset"
```

Reply:
Think I fixed the test. Waiting on build right now.